services/middlewareServices/AuthMiddlewareService.js

```diff
@@ -0,0 +1,102 @@
+// Import logger
+const logger = require("@logger/logger")
+
+// Import errors
+const { AuthError } = require("@errors/AuthError")
+
+const jwtUtils = require("@utils/jwt-utils")
+
+const { BadRequestError } = require("@root/errors/BadRequestError")
+
+const { E2E_TEST_REPO, E2E_TEST_SECRET, E2E_TEST_GH_TOKEN } = process.env
+const E2E_TEST_USER = "e2e-test"
+const GENERAL_ACCESS_PATHS = ["/v1/sites", "/v1/auth/whoami"]
+
+class AuthMiddlewareService {
+  verifyE2E({ cookies, url }) {
+    const { isomercmsE2E } = cookies
+    const urlTokens = url.split("/") // urls take the form "/v1/sites/<repo>/<path>""
+    let isValidE2E
+
+    if (isomercmsE2E) {
+      if (isomercmsE2E !== E2E_TEST_SECRET)
+        throw new AuthError("Bad credentials")
+
+      // Throw an error if accessing a repo other than e2e-test-repo
+      // Otherwise, allow access only to paths available to all users
+      if (!GENERAL_ACCESS_PATHS.includes(url)) {
+        if (urlTokens.length >= 3) {
+          const repo = urlTokens[3]
+          if (repo !== E2E_TEST_REPO)
+            throw new AuthError(
+              `E2E tests can only access the ${E2E_TEST_REPO} repo`
+            )
+        } else {
+          throw new BadRequestError("Invalid path")
+        }
+      }
+
+      isValidE2E = true
+    }
+
+    return isValidE2E
+  }
+
+  verifyJwt({ cookies, url }) {
+    const { isomercms } = cookies
+    const isValidE2E = this.verifyE2E({ cookies, url })
+
+    let accessToken
+    let userId
+    if (isValidE2E) {
+      accessToken = E2E_TEST_GH_TOKEN
+      userId = E2E_TEST_USER
+    } else {
+      try {
+        const {
+          access_token: retrievedToken,
+          user_id: retrievedId,
+        } = jwtUtils.verifyToken(isomercms)
+        accessToken = jwtUtils.decryptToken(retrievedToken)
+        userId = retrievedId
+      } catch (err) {
+        logger.error("Authentication error")
```

Comment:
i think we should have a more structured logging format to avoid information loss. for this, we could capture the cookie/url under a `params` key perhaps.

Reply:
Agreed, modified in f67202b, but we should have a further team discussion on modifying our existing error logging